### PR TITLE
Avoid calling irreps.dim and logger in forward (fix for #32)

### PR DESCRIPTION
**Pull request by mariogeiger**

### CODE REVIEW

**Comment by mariogeiger** [comment]
Fix for #32 

```python
import torch

import cuequivariance as cue
import cuequivariance_torch as cuet

m = cuet.FullyConnectedTensorProduct(
    irreps_in1=cue.Irreps("O3", "32x0e + 32x1o"),
    irreps_in2=cue.Irreps("O3", "32x0e + 32x1o"),
    irreps_out=cue.Irreps("O3", "32x0e + 32x1o"),
    layout=cue.mul_ir,
    optimize_fallback=True,
)

m_compile = torch.compile(m, fullgraph=True)
input1 = torch.randn(100, m.irreps_in1.dim)
input2 = torch.randn(100, m.irreps_in2.dim)
m_compile(input1, input2)
```
